packages/core-components/src/components/SideBarMobile.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+type SideBarMobileProps = {
+  closeButtonIcon: React.ReactNode;
+  onCloseBtnClick: () => void;
+  onSideBarItemClick: () => void;
+  dropDownIcon?: React.ReactNode;
+  sideBarItems: SideBarMobileItemProps[];
+};
+
+export type SideBarMobileItemProps = {
+  title: string;
+  url: string;
+  dropDownIcon?: React.ReactNode;
+  onSideBarItemClick?: () => void;
+  subItems?: SideBarMobileItemProps[];
+};
+
+const SideBarMobileItem: React.FC<SideBarMobileItemProps> = ({
+  title,
+  onSideBarItemClick,
+  dropDownIcon,
+  subItems,
+}) => {
+  const [itemOpen, setItemOpen] = React.useState(false);
+
+  const openSubItems = () => {
+    setItemOpen(!itemOpen);
+  };
+
+  const renderSubItems = () => {
+    if (subItems && subItems.length > 0 && itemOpen) {
+      return (
+        <ul className={`pl-6`}>
+          {subItems.map((subItem) => (
+            <SideBarMobileItem
+              key={subItem.title}
+              dropDownIcon={dropDownIcon}
+              onSideBarItemClick={onSideBarItemClick}
+              {...subItem}
+            />
+          ))}
+        </ul>
+      );
+    }
+    return null;
+  };
+
+  return (
+    <div className='w-full text-xl justify-between font-primary font-semibold my-5 text-white'>
+      {subItems ? (
+        <button
+          onClick={openSubItems}
+          className='flex w-full items-center justify-between'
+        >
+          {title}
+          <i
+            className={`w-10 h-10 transition-transform ${
+              itemOpen ? 'rotate-90' : ''
+            }`}
+          >
+            {dropDownIcon}
+          </i>
+        </button>
+      ) : (
+        <button onClick={onSideBarItemClick}>{title}</button>
+      )}
+      {renderSubItems()}
+    </div>
+  );
+};
+
+const SideBarMobile: React.FC<SideBarMobileProps> = ({
+  onCloseBtnClick,
+  onSideBarItemClick,
+  closeButtonIcon,
+  sideBarItems,
+  dropDownIcon,
+}) => {
+  return (
```

Comment:
just wonder, why do we have border-radius for this, this looks weird to me

Reply:
I see it like this. Does this mean the border-radius ?1

![1](https://github.com/its-rav/namvietstore-fe/assets/92157056/5521d6b3-6324-47af-808c-f345875274a7)

Reply:
nope. The design is using iphone layout I think, It should not be rounded